ygot/pathstrings.go

```diff
@@ -67,29 +67,43 @@ func PathToStrings(path *gnmipb.Path) ([]string, error) {
 
 	for i, e := range path.Elem {
 		if e.Name == "" {
-			return nil, fmt.Errorf("nil name for PathElem at index %d", i)
+			return nil, fmt.Errorf("empty name for PathElem at index %d", i)
 		}
 
-		elem := e.Name
-		if len(e.Key) != 0 {
-			var keys []string
-			for k, v := range e.Key {
-				if k == "" {
-					return nil, fmt.Errorf("nil key name (value: %s) in element %s", v, e.Name)
-				}
-				keys = append(keys, k)
-			}
-			sort.Strings(keys)
-
-			for _, k := range keys {
-				elem = fmt.Sprintf("%s[%s=%s]", elem, k, e.Key[k])
-			}
+		elem, err := elemToString(e.Name, e.Key)
+		if err != nil {
+			return nil, fmt.Errorf("failed formatting PathElem at index %d: %v", i, err)
 		}
 		p = append(p, elem)
 	}
 	return p, nil
 }
 
+func elemToString(name string, kv map[string]string) (string, error) {
+	if name == "" {
+		return "", errors.New("empty name for PathElem")
+	}
+	if len(kv) == 0 {
+		return name, nil
+	}
+
+	var keys []string
+	for k, v := range kv {
+		if k == "" {
+			return "", fmt.Errorf("empty key name (value: %s) in element %s", v, name)
+		}
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		v := strings.Replace(kv[k], `=`, `\=`, -1)
```

Comment:
I think the previous version of the specification said that `]` also needed to be escaped, see here: https://github.com/openconfig/reference/blob/792cdfbbe13cc8fbd3a20171671c95fe32e10fdb/rpc/gnmi/gnmi-path-conventions.md#paths-referencing-list-elements

Reply:
Done